sycl/include/sycl/reduction.hpp

```diff
@@ -1082,29 +1082,51 @@ struct NDRangeReduction<
   }
 };
 
-template <typename LocalRedsTy, typename BinOpTy, typename BarrierTy,
-          typename IdentityTy>
-void doTreeReduction(size_t WGSize, size_t LID, bool DisableExtraElem,
-                     IdentityTy Identity, LocalRedsTy &LocalReds, BinOpTy &BOp,
-                     BarrierTy Barrier) {
-  // For work-groups, which size is not power of two, local accessors have
-  // an additional element with index WGSize that is used by the
-  // tree-reduction algorithm. Initialize those additional elements with
-  // identity values here.
-  if (!DisableExtraElem)
-    LocalReds[WGSize] = Identity;
+/// Computes the greatest power-of-two less than or equal to N.
+static inline size_t GreatestPowerOfTwo(size_t N) {
+  if (N == 0)
+    return 0;
+
+  size_t Ret = 1;
+  while ((N >>= 1) != 0)
+    Ret <<= 1;
+  return Ret;
+}
+
```

Comment:
Can we use `sycl::detail::getNextPowerOfTwo` from https://github.com/intel/llvm/blob/d6f5b35286e5df3467ad73dd6734e15a1aa08c5c/sycl/include/sycl/detail/common.hpp#L427 instead?

Reply:
Discussed offline; we cannot use getNxtPowerOfTwo since it uses recursion. We could duplicate it and rewrite it as a loop, but we cannot change the existing implementation as it is constexpr.